### PR TITLE
OpenSSL 1.1.0 compatibility

### PR DESCRIPTION
**Pull request by greghudson**
This PR supersedes PR #442.  I have added a PKINIT RSA test case and split up the OpenSSL 1.1.0 compatibility changes into three commits: libk5crypto, k5tls, and PKINIT.

### CODE REVIEW

**Comment by greghudson** [comment]
For IP reasons, I have removed all of the changes from PR #442, leaving only my own work.  Of course this means the branch no longer builds against OpenSSL 1.1.  At some future time when I haven't looked at the PR #442 changes recently, I will redo that part of the work (or someone else can).

**Comment by greghudson** [comment]
Okay, I have redone the work and this is again ready for review.  This is now based on PR #466, so only look at the last three commits.

**Comment by greghudson** [comment]
Updated to take into account some upstream constification in DH_get0_pqg() and DH_get0_key(), and the deprecation of ASN1_STRING_data() in favor of ASN1_STRING_get0_data().

**Comment by greghudson** [comment]
OpenSSL 1.1.0 has been released.  I've verified that this branch still builds against it without warnings and passes tests.

**Comment by frozencemetery** [comment]
Confirmed it builds and passes tests on rhel7 (openssl-1.0.1e).  Also looks good to me.

File: src/configure.in [hunk 1]
```diff
@@ -1073,8 +1073,8 @@ enable_pkinit=try)
 if test "$enable_pkinit" = yes || test "$enable_pkinit" = try; then
   AC_CACHE_CHECK(for a recent enough OpenSSL, k5_cv_openssl_version_okay,
 [AC_COMPILE_IFELSE([AC_LANG_SOURCE([#include <openssl/opensslv.h>
-#if OPENSSL_VERSION_NUMBER < 0x00908000L
-# error openssl is too old, need 0.9.8
+#if OPENSSL_VERSION_NUMBER < 0x10000000L
+# error openssl is too old, need 1.0.0
```
Comment (tlyu) [inline src/configure.in]: Is this OpenSSL 1.0.0 minimum version requirement related to PKINIT, or is it the previously unchecked requirement related to CTS mode?

Comment (greghudson) [inline src/configure.in]: I wasn't able to find prior discussion of this (which surprises me), but X509_STORE_set_verify_cb() was added in OpenSSL 1.0, so the PKINIT code no longer builds against 0.9 with these changes.

**Comment by tlyu** [comment]
I like the refactoring of the DH parameter setup code.

**Pull request merged**